### PR TITLE
switch linting to ruff

### PR DESCRIPTION
**Pull request by reaperhulk**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -41,16 +41,15 @@ commands =
...
     python -m twisted.trial -j4 --reporter=text twisted
 
-[testenv:flake8]
+[testenv:lint]
 basepython = python3
 deps =
     black
-    flake8
-    flake8-import-order
+    ruff==0.0.284
```
Comment (alex) [inline tox.ini:48]: This is literally our only pin, and we have no mechanism for bumping. We need to either unpin or do this in a dependabot friendly way IMO.

Comment (reaperhulk) [reply tox.ini:48]: We have a pin for mypy in the exact same way here. It definitely isn't dependabot friendly though, I agree 😄

Comment (alex) [reply tox.ini:48]: Its also not local dev friendly (since I know neither of us tox)

**Review by alex — APPROVED** [review]
Going to merge this to get us back to green

**Pull request merged**